Index: .idea/workspace.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/workspace.xml b/.idea/workspace.xml
--- a/.idea/workspace.xml	
+++ b/.idea/workspace.xml	
@@ -3,9 +3,7 @@
   <component name="ChangeListManager">
     <list default="true" id="f530f221-4cb0-456e-a942-33dae9eb427f" name="Changes" comment="fix: исправить путь в index.css">
       <change beforePath="$PROJECT_DIR$/.idea/workspace.xml" beforeDir="false" afterPath="$PROJECT_DIR$/.idea/workspace.xml" afterDir="false" />
-      <change beforePath="$PROJECT_DIR$/README.md" beforeDir="false" afterPath="$PROJECT_DIR$/README.md" afterDir="false" />
       <change beforePath="$PROJECT_DIR$/scripts/index.js" beforeDir="false" afterPath="$PROJECT_DIR$/scripts/index.js" afterDir="false" />
-      <change beforePath="$PROJECT_DIR$/scripts/validate.js" beforeDir="false" afterPath="$PROJECT_DIR$/scripts/validate.js" afterDir="false" />
     </list>
     <option name="SHOW_DIALOG" value="false" />
     <option name="HIGHLIGHT_CONFLICTS" value="true" />
Index: scripts/index.js
===================================================================
diff --git a/scripts/index.js b/scripts/index.js
--- a/scripts/index.js	
+++ b/scripts/index.js	
@@ -18,43 +18,33 @@
 const profileEditForm = popupProfileEdit.querySelector('.popup__form');
 
 // Функция управляет зыкрытием попапа по нажатию 'esc'
-function closePopupFromEsc(evt, popup) {
+function closePopupFromEsc(evt) {
+  const activePopup = document.querySelector('.popup_opened')
   if (evt.key === 'Escape') {
-    console.log('qwe')
-    closePopup(popup)
+    closePopup(activePopup)
   }
 }
 
 // Функция управляет зыкрытием попапа по клику на оверлей
-function closePopupFromOverlay() {
-  const popupList = Array.from(document.querySelectorAll('.popup'))
-  popupList.forEach(popup => {
-    popup.addEventListener('click', (evt) => {
-      closePopup(evt.target)
-    })
-  })
+function closePopupFromOverlay(evt) {
+  const activePopup = document.querySelector('.popup_opened')
+  if (evt.target.classList.contains('popup')) {
+    closePopup(activePopup)
+  }
 }
 
-closePopupFromOverlay()
-function callClosePopupFromEsc(evt) {
-  closePopupFromEsc(evt, popup)
-}
+
 // Функция управляет открытием попапа
 function openPopup(popup) {
-
-  document.addEventListener('keydown', callClosePopupFromEsc)
-  // document.addEventListener('keydown', (evt) => {
-  //   closePopupFromEsc(evt, popup)
-  // }, {once: true})
+  document.addEventListener('keydown', closePopupFromEsc)
+  popup.addEventListener('mousedown', closePopupFromOverlay)
   popup.classList.add('popup_opened'); //добавляем класс открытия
 }
 
 // Функция управляет закрытием попапа
 function closePopup(popup) {
-  // document.removeEventListener('keydown', (evt) => {
-  //   closePopupFromEsc(evt, popup)
-  // }, {once: true})
-  document.removeEventListener('keydown', callClosePopupFromEsc)
+  document.removeEventListener('keydown', closePopupFromEsc)
+  popup.removeEventListener('mousedown', closePopupFromOverlay)
   popup.classList.remove('popup_opened'); //убираем класс открытия
 }
 
